cmd/generate/config/rules/datafactory.go

```diff
@@ -0,0 +1,22 @@
+package rules
+
+import (
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func AzureDataFactory() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "Uncovered a Data Factory Self-Hosted Integration Runtime Key, which may compromise big data analytics platforms and sensitive data processing.",
+		RuleID:      "datafactory-shir-token",
+		// SHIR Key format: IR@{GUID}@{string_azure _resource_name}@{string_azure_resource_location}@{string_base64}
+		Regex:    generateUniqueTokenRegex(`IR@[0-9a-zA-Z-]{36}@(.*?)@[0-9a-zA-Z\-=]*@[A-Za-z0-9+\/=]{44}`, false),
+		Keywords: []string{"IR"},
```

Comment:
@RachidAZ Could `IR` become `IR@` or is the `@` not guaranteed?

Reply:
@zricethezav I am not sure whether `@` is guaranteed, `"IR"` should do the job :)

Reply:
More specificity ensures better performance. Please replace `IR` with `IR@` otherwise we could see a significant performance hit since `IR` is going to trigger lots of regex checks

Reply:
@zricethezav makes sense, updated. Thanks.